### ISSUE TITLE
How to make changes that affect `pyiron_atomistics`?

### ISSUE THREAD

**Issue by samwaseda** [pyiron/pyiron_base#970]
I'm making changes in [this PR](https://github.com/pyiron/pyiron_base/pull/952) and realized that I have to change some stuff in `pyiron_atomistics`. How can I make changes in multiple repositories without breaking anything?

**Comment by pmrv** [comment]
Imo make the change here, make a new release, wait for the dependabot PR on atomistics (which will fail the unit tests), then apply the changes in atomistics you need to accommodate the changes you made in base, then trigger a new release of atomistics.  This way there'll be no broken version of atomistics.

**Comment by samwaseda** [comment]
ah wait, there'd be a problem. So the thing is, in the PR that I mentioned above, I replace a `ValueError` by a `KeyError`. Now there are a few places in `pyiron_atomistics`, which catch `ValueError`, but not `KeyError`, meaning as soon as the new version is released, there would be a problem.

**Comment by pmrv** [comment]
So this is one of the reasons why I originally objected to unpinning dependencies upon uploading them to conda.  Imo since we specify `==` constraints in the repo this exact case of having to change version in two repositories together would not cause issues because until the atomistics version is bumped it would still depend on the old (working) version of base.

If you want to be exactly sure not to cause issues, change [this workflow](https://github.com/pyiron/pyiron_atomistics/blob/main/.github/workflows/deploy.yml) to not unpin `pyiron_*` deps (the line with `sed`), release a new version then do what I sketched above.  I'm not sure if this will automatically change the deps advertised to the conda package since they seem to be hardcoded [here](https://github.com/conda-forge/pyiron_atomistics-feedstock/blob/main/recipe/meta.yaml).

Imo the reasonable thing to do midterm is to change our exact dep constraints to `lower<,<upper` where we manually bump `lower` whenever we need newer stuff from a dependency and let dependabot bump (and test) `upper` and advertise these exact bounds to both pip and conda.

**Comment by niklassiemer** [comment]
Yes, in the mentioned workflow, only the pip dependencies are changed.

**Comment by liamhuber** [comment]
> If you want to be exactly sure not to cause issues, change [this workflow](https://github.com/pyiron/pyiron_atomistics/blob/main/.github/workflows/deploy.yml) to not unpin pyiron_* deps (the line with sed), release a new version then do what I sketched above. I'm not sure if this will automatically change the deps advertised to the conda package since they seem to be hardcoded [here](https://github.com/conda-forge/pyiron_atomistics-feedstock/blob/main/recipe/meta.yaml).

If we choose to do this, at the same time please go over to pyiron/actions and make a PR giving the same change(s) in the [centralized deployment workflow](https://github.com/pyiron/actions/blob/main/.github/workflows/release.yml)
